io/pci/EEH.py

```diff
@@ -261,3 +285,31 @@ def is_baremetal():
         if 'PowerNV' in genio.read_file("/proc/cpuinfo").strip():
             return True
         return False
+
+    def get_paths(self, fc_host):
+        '''
+        returns the list of paths coressponding to the given fc_host
+        '''
+        paths = []
+        split_pattern = r':|\.'
+        fc_host = ':'.join(re.split(split_pattern, fc_host)[0:3])
+        cmd = 'ls -l /sys/block/ | grep -i %s' % fc_host
+        for line in process.getoutput(cmd).split('\n'):
+            if "/%s" % fc_host in line:
+                paths.append(line.split("/")[-1])
+        return paths
+
+    def get_host_state(self, pci_devices):
+        '''
+        returns state of fc hosts corresponding to the given pci devices
+        '''
+        host_state = []
+        split_pattern = r':|\.'
+        pci_devices = ':'.join(re.split(split_pattern, pci_devices)[0:3])
+        cmd = 'ls -l /sys/class/fc_host | grep -i %s' % pci_devices
+        for line in process.getoutput(cmd).split('\n'):
+            if "/%s" % pci_devices in line:
+                host_state.append(process.getoutput(
+                'cat /sys/class/fc_host/%s/port_state' % line.split("/")[-1]))
+        return host_state
+
```

Comment:
check if we have common utils for these methods.

Reply:
Yeah! There is no method in common utils. I am planning to move it to under utils as it can be used with mulitple devices.

Reply:
even this as well